lib/libc/Kconfig

```diff
@@ -55,8 +55,8 @@ menu "C Library"
 choice LIBC_IMPLEMENTATION
 	prompt "C Library Implementation"
 	default EXTERNAL_LIBC if NATIVE_APPLICATION
+	default PICOLIBC
 	default NEWLIB_LIBC if REQUIRES_FULL_LIBC
-	default PICOLIBC if REQUIRES_FULL_LIBC
 	default MINIMAL_LIBC
```

Comment:
I'd say that we need to have better undestanding of our strategy about Zephyr minimal libc and newlib support.

* minimal libc:
As per #49922 picolibc is now supported on all Zephyr architectures (correct me if I'm wrong), so when this change will be applied we'll stop test anyting with minimal libc in upstream CI.

  As I understand we still have plans to keep Zephyr minimal libc - so we need it to have it tested in CI.

* newlib:
It's not clear if we are going to support (and do testing) both picolibc and newlib (for SDK's toolchain) simultaneously or we are going to switch for picolibc for both use cases (if minimal libc implementation is required and if full-sized libc required).

  As some tests select newlib directly in `prj.conf` by setting `CONFIG_NEWLIB_LIBC=y` and not `CONFIG_REQUIRES_FULL_LIBC=y` currently we'll still do some testing in upstream CI with newlib even after this PR is applied. Though currently it's not clear if we are going to keep it that way or we are going to modify these tests to select picolibc by default (as we already mark picolibc as a full-sized libc).

Reply:
> The idea is that the kernel tests will still build with the minimal libc to ensure that the Zephyr kernel can still build using the minimal libc even if we switch to Picolibc as the default -- this is to be implemented (tracked by "Devise a process to ensure that the Zephyr kernel can keep building with the minimal libc" in #49922).

Do we want to do this by creating explicit tests that select minimal libc (as is currently done for newlib and picolibc)? Or do we want a separate mechanism to enable another CI pipeline that runs tests against minimal libc instead of picolibc?

> That needs to be changed to use `CONFIG_REQUIRES_FULL_LIBC=y` unless the test/sample **must** use newlib.

We also have tests for libc functionality that currently run against all of the SDK C libraries. Expanding that to cover more of the C library functionality used by Zephyr would also serve to help get better C library testing for all of them.

Reply:
It sounds like we need to change some kernel tests to explicitly test against the minimal C library? I'd think those would want to test against both that and the default C library? That seems easy enough to manage.

Reply:
I've expanded the kernel tests to run against both the default C library as well as the minimal C library.